### PR TITLE
Move OAuth flow tests from feature tests to system tests

### PR DESCRIPTION
**Pull request by ClearlyClaire**
Attempt to run in these tests in a real browser, since browsers may possibly behave differently than Capybara regarding redirects and cookies.

Changed the callback address to `http://localhost/health` so that it's accessible to Capybara and different from the default redirect. Ideally, we want something entirely distinct from the Rails app here.

### CODE REVIEW

File: spec/system/oauth_spec.rb [hunk 1]
```diff
@@ -3,13 +3,16 @@
...
 
   context 'when the user is already logged in' do
     let!(:user) { Fabricate(:user) }
 
     before do
-      sign_in user, scope: :user
+      visit new_user_session_path
+      fill_in 'user_email', with: user.email
+      fill_in 'user_password', with: user.password
+      click_on I18n.t('auth.login')
```
Comment (mjankowski) [inline spec/system/oauth_spec.rb:15]: There's a helper method near bottom of file to wrap up the interaction of filling in and submitting the auth form.

Comment (ClearlyClaire) [reply spec/system/oauth_spec.rb:15]: It's not in scope!

**Comment by mjankowski** [comment]
> Changed the callback address to `http://localhost/health` so that it's accessible to Capybara and different from the default redirect. Ideally, we want something entirely distinct from the Rails app here.

I think this is fine and a net improvement to be distinct from the default case. Also agreed that a good further improvement would be to get the redirect check elsewhere (use some combination of the oauth test/spec mode and/or webmock). I also think if we're going to add more oauth edge case coverage we should contemplate which portions should be in request specs for the http level (redirects, response codes, headers, etc) and which should stay at the user/browser level in system specs. In this spec already, I think the check for the redirect uri (instead of something expected to be found on a stubbed page for that uri by the browser) is in a grey area. That said, the oauth flow is itself sort of a grey area of browser/backend.

**Review by mjankowski — APPROVED** [review]

**Review by renchap — APPROVED** [review]
